### PR TITLE
Tabs animated selection

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Current Behavior

Each tab draws its own selection indicator.
The selection indicator appears/disappears as each tab is selected.

## New Behavior

The tab list draws the selection indicator.
The selection indicator can be animated to slide as each tab is selected.

## Related Issue(s)

Updates #20338

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/077011a435e06cf1c4f5d39fce038846282fcd69'>077011a435e06cf1c4f5d39fce038846282fcd69</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ContextualMenu-mount_base.html">14831</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ContextualMenu-mount.html">7255</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ContextualMenu-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-mount_base.html">121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-mount.html">178</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Avatar-mount_base.html">830</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Avatar-mount.html">799</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/BaseButton-mount_base.html">793</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/BaseButton-mount.html">812</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Breadcrumb-mount_base.html">2278</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Breadcrumb-mount.html">2308</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ButtonNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ButtonNext-mount_base.html">444</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ButtonNext-mount.html">449</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Checkbox-mount_base.html">1343</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Checkbox-mount.html">1339</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/CheckboxBase-mount_base.html">1117</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/CheckboxBase-mount.html">1129</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ChoiceGroup-mount_base.html">4121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ChoiceGroup-mount.html">4134</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ComboBox-mount_base.html">874</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ComboBox-mount.html">832</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/CommandBar-mount_base.html">8803</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/CommandBar-mount.html">8846</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ContextualMenu-mount_base.html">14831</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ContextualMenu-mount.html">7255</a></td>
            <td>1000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ContextualMenu-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DefaultButton-mount_base.html">982</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DefaultButton-mount.html">1003</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DetailsRow-mount_base.html">3240</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DetailsRow-mount.html">3249</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DetailsRowFast-mount_base.html">3273</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DetailsRowFast-mount.html">3265</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DetailsRowNoStyles-mount_base.html">3153</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DetailsRowNoStyles-mount.html">3055</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Dialog-mount_base.html">2244</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Dialog-mount.html">2245</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DocumentCardTitle-mount_base.html">167</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/DocumentCardTitle-mount.html">163</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Dropdown-mount_base.html">2779</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Dropdown-mount.html">2817</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderNext-mount_base.html">1665</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderNext-mount.html">1667</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-mount_base.html">121</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-mount.html">178</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-virtual-rerender_base.html">104</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-virtual-rerender.html">90</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">188</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FluentProviderWithTheme-virtual-rerender-with-unmount.html">181</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FocusTrapZone-mount_base.html">1591</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FocusTrapZone-mount.html">1599</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FocusZone-mount_base.html">1582</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/FocusZone-mount.html">1612</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/IconButton-mount_base.html">1507</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/IconButton-mount.html">1510</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Label-mount_base.html">335</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Label-mount.html">333</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Layer-mount_base.html">2554</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Layer-mount.html">2576</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Link-mount_base.html">424</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Link-mount.html">440</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/MakeStyles-mount_base.html">1515</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/MakeStyles-mount.html">1490</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/MenuButton-mount_base.html">1286</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/MenuButton-mount.html">1281</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/MessageBar-mount_base.html">1747</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/MessageBar-mount.html">1774</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Nav-mount_base.html">2862</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Nav-mount.html">2848</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/OverflowSet-mount_base.html">977</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/OverflowSet-mount.html">993</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Panel-mount_base.html">2135</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Panel-mount.html">2192</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Persona-mount_base.html">765</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Persona-mount.html">755</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Pivot-mount_base.html">1256</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Pivot-mount.html">1257</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/PrimaryButton-mount_base.html">1128</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/PrimaryButton-mount.html">1137</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Rating-mount_base.html">6714</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Rating-mount.html">6669</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/SearchBox-mount_base.html">1175</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/SearchBox-mount.html">1158</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Shimmer-mount_base.html">2240</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Shimmer-mount.html">2183</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Slider-mount_base.html">1702</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Slider-mount.html">1711</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/SpinButton-mount_base.html">4265</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/SpinButton-mount.html">4309</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Spinner-mount_base.html">406</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Spinner-mount.html">403</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/SplitButton-mount_base.html">2717</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/SplitButton-mount.html">2749</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Stack-mount_base.html">466</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Stack-mount.html">471</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/StackWithIntrinsicChildren-mount_base.html">1991</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/StackWithIntrinsicChildren-mount.html">2017</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/StackWithTextChildren-mount_base.html">4556</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/StackWithTextChildren-mount.html">4525</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/SwatchColorPicker-mount_base.html">9911</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/SwatchColorPicker-mount.html">10107</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/TagPicker-mount_base.html">2298</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/TagPicker-mount.html">2315</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/TeachingBubble-mount_base.html">11143</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/TeachingBubble-mount.html">10911</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Text-mount_base.html">390</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Text-mount.html">400</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/TextField-mount_base.html">1208</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/TextField-mount.html">1238</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ThemeProvider-mount_base.html">993</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ThemeProvider-mount.html">1061</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ThemeProvider-virtual-rerender_base.html">551</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ThemeProvider-virtual-rerender.html">544</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1594</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1615</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Toggle-mount_base.html">722</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/Toggle-mount.html">692</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/buttonNative-mount_base.html">135</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21507/perf-test/buttonNative-mount.html">128</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

## Perf Analysis (`@fluentui/react-northstar`)
:warning: No perf measurements available
<div id="perfComment9423"/>

File: packages/react-tabs/src/tab.constants.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// TODO: These constants should be replaced with design tokens
```
Comment (micahgodbolt) [inline packages/react-tabs/src/tab.constants.ts:1]: is this TODO being punted for now?

File: packages/react-tabs/src/components/TabList/useTabListStyles.ts [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
 import type { TabListState } from './TabList.types';
+import { tokens } from '@fluentui/react-theme';
+import { usePrevious } from '@fluentui/react-utilities';
+import { tabPendingDesignTokens } from '../../tab.constants';
 
 export const tabListClassName = 'fui-TabList';
+export const tabListSelectionIndicatorName = 'fui-TabList_SelectionIndicator';
+
+export const indicatorOffsetVar = '--selection-indicator-offset';
```
Comment (micahgodbolt) [inline packages/react-tabs/src/components/TabList/useTabListStyles.ts:10]: these should use your prefix to ensure that there aren't any possible conflicts.

```suggestion
export const indicatorOffsetVar = '--fui-TabList-selection-indicator-offset';
```

File: packages/react-tabs/src/components/TabList/useTabListStyles.ts [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
+export const indicatorLengthVar = '--selection-indicator-length';
```
Comment (micahgodbolt) [inline packages/react-tabs/src/components/TabList/useTabListStyles.ts:11]:
```suggestion
export const indicatorLengthVar = '--fui-Tablist-selection-indicator-length';
```

File: packages/react-tabs/src/components/TabList/useTabListStyles.ts [hunk 4]
```diff
@@ -10,22 +17,115 @@ const useStyles = makeStyles({
...
   },
   vertical: {
     flexDirection: 'column',
   },
 });
 
+/**
+ * Styles for the selection indicator slot when horizontal.
+ */
+const useHorizontalIndicatorStyles = makeStyles({
```
Comment (micahgodbolt) [inline packages/react-tabs/src/components/TabList/useTabListStyles.ts:30]: possible todo for later: horizontal and vertical styles have a ton in common. could be combined with horizontal/vertical being state just like small/animated

**Review by micahgodbolt — APPROVED** [review]

File: packages/react-tabs/src/tab.constants.ts (hunk 1 above)
Comment (GeoffCoxMSFT) [reply packages/react-tabs/src/tab.constants.ts:1]: Yes - other controls like button has spacings that aren't defined in the design tokens, but are specified in the figma.

File: packages/react-tabs/src/components/TabList/useTabListStyles.ts (hunk 2 above)
Comment (GeoffCoxMSFT) [reply packages/react-tabs/src/components/TabList/useTabListStyles.ts:10]: I don't think css vars need the prefix - they are setting the value within the component and aren't meant to be addressable like the style names.  We could ask for people's thoughts at tech sync.

File: packages/react-tabs/src/components/TabList/useTabListStyles.ts (hunk 4 above)
Comment (GeoffCoxMSFT) [reply packages/react-tabs/src/components/TabList/useTabListStyles.ts:30]: The problem is that one is :before and one is :after.  I couldn't figure out a way to re-use common styles for both tags.  I didn't want to rely on the content being unset and applying a bunch of dummy styles to both.  Interested in hearing ways to do this :)

**Pull request merged**